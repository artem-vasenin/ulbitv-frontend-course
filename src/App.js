import './App.css';

function App() {
  return (
    <div className="App">
      <h1>Hello Rusich</h1>
      <button>Click</button>
      <input type="text" placeholder="password"/>
    </div>
  );
}

export default App;
