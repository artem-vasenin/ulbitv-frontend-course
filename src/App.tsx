import React from 'react';
import { useAppDispatch, useAppSelector } from './hooks/redux';
import { userSlice } from './store/reducers/userSlice';

function App() {
  const { count } = useAppSelector(state => state.userReducer);
  const { inc } = userSlice.actions;
  const dispatch = useAppDispatch();

  return (
    <div className="App">
      <h1>{count}</h1>
      <button onClick={() => dispatch(inc(1))}>INC</button>
    </div>
  );
}

export default App;
