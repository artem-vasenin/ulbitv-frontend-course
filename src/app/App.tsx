import { Suspense } from 'react';
import { Link, Route, Routes } from 'react-router-dom';

import { useTheme } from 'app/providers/ThemeProvider';
import { AboutPage } from 'pages/About';
import { MainPage } from 'pages/MainPage';
import { classNames } from 'shared/lib/classNames/classNames';
import './styles/index.scss';

export const App = () => {
  const { theme, toggleTheme } = useTheme();

  return (
    <div className={classNames('app', [theme])}>
      <div>
        <Link to={'/'}>Home</Link>
      </div>
      <div>
        <Link to={'/about'}>About</Link>
      </div>

      <button onClick={toggleTheme}>Theme</button>

      <Suspense fallback={<div>Loading...</div>}>
        <Routes>
          <Route path={'/about'} element={<AboutPage/>}/>
          <Route path={'/'} element={<MainPage/>}/>
        </Routes>
      </Suspense>
    </div>
  );
};
