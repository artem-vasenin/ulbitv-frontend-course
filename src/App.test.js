import { render, screen } from '@testing-library/react';
import App from './App';

describe('Test App', () => {
  test('renders react title', () => {
    render(<App />);
    const titleElement = screen.getByText(/hello rusich/i);
    const btnElement = screen.getByRole('button');
    const inputElement = screen.getByPlaceholderText(/password/i);
    expect(titleElement).toBeInTheDocument();
    expect(btnElement).toBeInTheDocument();
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toMatchSnapshot();
  });
});
