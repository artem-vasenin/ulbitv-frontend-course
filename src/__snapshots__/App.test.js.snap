// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App renders react title 1`] = `
<input
  placeholder="password"
  type="text"
/>
`;

exports[`Test App renders react title 2`] = `
<input
  placeholder="password"
  type="text"
/>
`;

exports[`renders react title 1`] = `
<input
  placeholder="password"
  type="text"
/>
`;
